packages/client-sdk-nodejs/src/internal/grpc/client-timeout-interceptor.ts

```diff
@@ -1,14 +1,123 @@
-import {InterceptingCall, Interceptor} from '@grpc/grpc-js';
+import {MomentoLogger, MomentoLoggerFactory} from '@gomomento/sdk-core';
+import {InterceptingCall, Interceptor, status} from '@grpc/grpc-js';
+import {RetryStrategy} from '../../config/retry/retry-strategy';
+import {DefaultStorageRetryStrategy} from '../../config/retry/storage-default-retry-strategy';
+import {DefaultMomentoLoggerFactory} from '../../config/logging/default-momento-logger';
 
+// Determine which retry strategy is specified in the configuration
+// and which interceptor to use.
 export const ClientTimeoutInterceptor = (
-  requestTimeoutMs: number
+  overallRequestTimeoutMs: number,
+  retryStrategy?: RetryStrategy,
+  loggerFactory?: MomentoLoggerFactory
 ): Interceptor => {
-  return (options, nextCall) => {
-    if (!options.deadline) {
-      const deadline = new Date(Date.now());
-      deadline.setMilliseconds(deadline.getMilliseconds() + requestTimeoutMs);
-      options.deadline = deadline;
-    }
-    return new InterceptingCall(nextCall(options));
-  };
+  if (
+    retryStrategy !== undefined &&
+    retryStrategy instanceof DefaultStorageRetryStrategy
+  ) {
+    const responseDataReceivedTimeoutMs =
+      retryStrategy.getResponseDataReceivedTimeoutMillis();
+    return new RetryUntilTimeoutInterceptor(
+      loggerFactory ?? new DefaultMomentoLoggerFactory(),
+      overallRequestTimeoutMs,
+      responseDataReceivedTimeoutMs
+    ).createTimeoutInterceptor();
+  }
+  return new BasicTimeoutInterceptor(
+    overallRequestTimeoutMs
+  ).createTimeoutInterceptor();
 };
+
+export class BasicTimeoutInterceptor {
```

Comment:
I don't think these need to be exported, hopefully they are never used/visible outside of this file, right?